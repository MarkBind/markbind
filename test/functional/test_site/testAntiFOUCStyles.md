<!-- Anti-FOUC styles should be applied to navbar and dropdown in navbar. -->

<navbar type="dark">
  <li><a href="/" class="nav-link">One</a></li>
  <li><a href="/" class="nav-link">Two</a></li>
  <dropdown header="Dropdown" class="nav-link">
    <li><a class="dropdown-item" href="/">Dropdown One</a></li>
    <li><a class="dropdown-item" href="/">Dropdown Two</a></li>
  </dropdown>
</navbar>

<!-- Anti-FOUC styles should be applied to dropdown. -->
**Test dropdown in body with text and class attributes**

<dropdown header="Test One" class="test-class">
  <li><a class="dropdown-item" href="/">Dropdown One</a></li>
  <li><a class="dropdown-item" href="/">Dropdown Two</a></li>
</dropdown>

<!-- Anti-FOUC styles should be correctly applied to dropdown with no text or class. -->

**Test dropdown in body without text and class attributes**

<dropdown>
  <button slot="button" type="button" class="btn dropdown-toggle">
    Test Two
    <span class="caret"></span>
  </button>
  <li><a class="dropdown-item" href="/">Dropdown One</a></li>
  <li><a class="dropdown-item" href="/">Dropdown Two</a></li>
</dropdown>

<!-- Filler text to increase page length. -->

**Filler text**

<br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/>

Some text
