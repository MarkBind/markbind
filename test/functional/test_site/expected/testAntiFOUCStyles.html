<!DOCTYPE html>
<html>
<head>
    <meta name="default-head-top">
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="generator" content="MarkBind 1.22.0">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>Hello World</title>
    <link rel="stylesheet" href="markbind/css/bootstrap.min.css">
    <link rel="stylesheet" href="markbind/css/bootstrap-vue.min.css">
    <link rel="stylesheet" href="markbind/fontawesome/css/all.min.css" >
    <link rel="stylesheet" href="markbind/glyphicons/css/bootstrap-glyphicons.min.css" >
    <link rel="stylesheet" href="markbind/css/github.min.css">
    <link rel="stylesheet" href="markbind/css/markbind.css">
    <link rel="stylesheet" href="STYLESHEET_LINK">
    <link rel="stylesheet" href="markbind/layouts/default/styles.css">
    
    
    <meta name="default-head-bottom">
    <link rel="icon" href="/test_site/favicon.png">
</head>
<body>
<div id="app">
    <div id="content-wrapper">

  <navbar placement="top" type="inverse" class="temp-navbar">
    <li><a href="/" class="nav-link">One</a></li>
    <li><a href="/" class="nav-link">Two</a></li>
    <div class="nav-link temp-dropdown-placeholder">Dropdown</div>
    <dropdown text="Dropdown" class="nav-link temp-dropdown">
      <li><a class="dropdown-item" href="/">Dropdown One</a></li>
      <li><a class="dropdown-item" href="/">Dropdown Two</a></li>
    </dropdown>
  </navbar>

  <h1 id="test-dropdown-in-body-with-text-and-class-attributes">Test dropdown in body with text and class attributes<a class="fa fa-anchor" href="#test-dropdown-in-body-with-text-and-class-attributes"></a></h1>
  <div class="test-class temp-dropdown-placeholder">Test One</div>
  <dropdown text="Test One" class="test-class temp-dropdown">
    <li><a class="dropdown-item" href="/">Dropdown One</a></li>
    <li><a class="dropdown-item" href="/">Dropdown Two</a></li>
  </dropdown>

  <h1 id="test-dropdown-in-body-without-text-and-class-attributes">Test dropdown in body without text and class attributes<a class="fa fa-anchor" href="#test-dropdown-in-body-without-text-and-class-attributes"></a></h1>
  <div class="temp-dropdown-placeholder"></div>
  <dropdown class="temp-dropdown">
    <button slot="button" type="button" class="btn dropdown-toggle">
    Test Two
    <span class="caret"></span></button>
    <li><a class="dropdown-item" href="/">Dropdown One</a></li>
    <li><a class="dropdown-item" href="/">Dropdown Two</a></li>
  </dropdown>

  <h1 id="filler-text">Filler text<a class="fa fa-anchor" href="#filler-text"></a></h1>
  <p><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br></p>
  <p>Some text</p>
</div>
<div id="flex-div"></div>
<footer>
  <div class="text-center">
    Default footer
  </div>
</footer>
</div>
</body>
<script src="markbind/js/vue.min.js"></script>
<script src="markbind/js/vue-strap.min.js"></script>
<script src="markbind/js/bootstrap-utility.min.js"></script>
<script src="markbind/js/polyfill.min.js"></script>
<script src="markbind/js/bootstrap-vue.min.js"></script>
<script>
    const baseUrl = '/test_site'
    const enableSearch = true
</script>
<script src="markbind/js/setup.js"></script>
<script src="SCRIPT_LINK"></script>
<script>alert("hello")</script>
<script src="markbind/layouts/default/scripts.js"></script>
</html>
